test/e2e/scoped_client_test.go

```diff
@@ -47,80 +62,82 @@ func TestScopedClient(t *testing.T) {
+		// 4. Get scoped client instance(s)
+		// 5. Invoke Get API call on non existent object(s) to check if
+		//    the call can be made successfully.
+		namespace := genName("a")
+		_, cleanupNS := newNamespace(GinkgoT(), kubeclient, namespace)
+		defer cleanupNS()
+
+		saName := genName("user-defined-")
+		sa, cleanupSA := newServiceAccount(GinkgoT(), kubeclient, namespace, saName)
+		defer cleanupSA()
+
+		waitForServiceAccountSecretAvailable(GinkgoT(), kubeclient, sa.GetNamespace(), sa.GetName())
+
+		strategy := scoped.NewClientAttenuator(logger, config, kubeclient, crclient, dynamicclient)
+		getter := func() (reference *corev1.ObjectReference, err error) {
+			reference = &corev1.ObjectReference{
+				Namespace: namespace,
+				Name:      saName,
 			}
 
-			if tt.grant != nil {
-				cleanupPerm := tt.grant(sa.GetNamespace(), sa.GetName())
-				defer cleanupPerm()
-			}
+			return
+		}
 
-			// We expect to get scoped client instance(s).
-			kubeclientGot, crclientGot, dynamicClientGot, errGot := strategy.AttenuateClient(getter)
-			require.NoError(t, errGot)
-			require.NotNil(t, kubeclientGot)
-			require.NotNil(t, crclientGot)
-
-			_, errGot = kubeclientGot.KubernetesInterface().CoreV1().ConfigMaps(namespace).Get(genName("does-not-exist-"), metav1.GetOptions{})
-			require.Error(t, errGot)
-			tt.assertFunc(errGot)
-
-			_, errGot = crclientGot.OperatorsV1alpha1().CatalogSources(namespace).Get(genName("does-not-exist-"), metav1.GetOptions{})
-			require.Error(t, errGot)
-			tt.assertFunc(errGot)
-
-			gvr := schema.GroupVersionResource{Group: "", Version: "v1", Resource: "ConfigMap"}
-			_, errGot = dynamicClientGot.Resource(gvr).Namespace(namespace).Get(genName("does-not-exist-"), metav1.GetOptions{})
-			require.Error(t, errGot)
-			tt.assertFunc(errGot)
-		})
-	}
-}
+		if tt.grant != nil {
+			cleanupPerm := tt.grant(sa.GetNamespace(), sa.GetName())
+			defer cleanupPerm()
+		}
+
+		// We expect to get scoped client instance(s).
+		kubeclientGot, crclientGot, dynamicClientGot, errGot := strategy.AttenuateClient(getter)
+		require.NoError(GinkgoT(), errGot)
+		require.NotNil(GinkgoT(), kubeclientGot)
+		require.NotNil(GinkgoT(), crclientGot)
+
+		_, errGot = kubeclientGot.KubernetesInterface().CoreV1().ConfigMaps(namespace).Get(genName("does-not-exist-"), metav1.GetOptions{})
+		require.Error(GinkgoT(), errGot)
+		tt.assertFunc(errGot)
+
+		_, errGot = crclientGot.OperatorsV1alpha1().CatalogSources(namespace).Get(genName("does-not-exist-"), metav1.GetOptions{})
+		require.Error(GinkgoT(), errGot)
+		tt.assertFunc(errGot)
+
+		gvr := schema.GroupVersionResource{Group: "", Version: "v1", Resource: "ConfigMap"}
+		_, errGot = dynamicClientGot.Resource(gvr).Namespace(namespace).Get(genName("does-not-exist-"), metav1.GetOptions{})
+		require.Error(GinkgoT(), errGot)
+		tt.assertFunc(errGot)
+	},
+
+		table.Entry(tests[0].name, tests[0]),
+		table.Entry(tests[0].name, tests[0]),
```

Comment:
This will run the same case twice (looks like a copy-paste mistake). It might be best to replace these slice index expressions with the test values themselves to prevent the same mistake from being introduced in the future, or construct a slice of table.Entry and pass it as varargs to DescribeTable.

Reply:
Modified the test to remove `tests[]` slice and instead use `tableEntries[]` of type `Table.Entry`